src/NUnitEngine/nunit-agent/AgentExitCodes.cs

```diff
@@ -0,0 +1,36 @@
+﻿// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+namespace NUnit.Common
+{
+    internal static class AgentExitCodes
+    {
+        public const int OK = 0;
+        public const int PARENT_PROCESS_TERMINATED = -1;
+        public const int UNEXPECTED_EXCEPTION = -2;
+        public const int FAILED_TO_START_AGENT = -3;
+        public const int NO_DEBUGGER_SECURITY = -4;
+        public const int NO_DEBUGGER_NOT_IMPLEMENTED = -5;
```

Comment:
I didn't know about these problems. Are there issues for them? Of course, it's unlikely users will ever see these codes, since `--debug-agent` is for our debugging, but it could save time if we get a sensible message.

How about DEBUGGER_SECURITY_VIOLATION and DEBUGGER_NOT_IMPLEMENTED.

Also, we should have a page in the wiki that lists these codes.

Reply:
Yes, the --debug option allows you to debug tests, even if you are running on a separate agent. The setting is passed in the package. The --debug-agent option is just for us to debug the agent.